features/301.ngsild.TimeseriesData.feature

```diff
@@ -0,0 +1,121 @@
+    Scenario Outline: Using Quantum Leap API list the first 3 sampled values
+    When I prepare a GET HTTP request to "<url>"
+    And  I set header fiware-service to openiot
+    And  I set header fiware-servicepath to /
+    And  I set header accept to application/json
+    And  I perform the query request
+    Then  I receive a HTTP "200" response code
+    Examples:
+      | url                                                                                            |
+      | http://localhost:8668/v2/entities/urn:ngsi-ld:Device:filling001/attrs/filling?limit=3          |
+      | http://localhost:8668/v2/entities/urn:ngsi-ld:Device:filling001/attrs/filling?offset=3&limit=3 |
+      | http://localhost:8668/v2/entities/urn:ngsi-ld:Device:filling001/attrs/filling?lastN=3          |
+      | http://localhost:8668/v2/entities/urn:ngsi-ld:Device:filling001/attrs/filling?aggrMethod=count&aggrPeriod=minute&lastN=3 |
+      | http://localhost:8668/v2/entities/urn:ngsi-ld:Device:filling001/attrs/filling?aggrMethod=min&aggrPeriod=minute&lastN=3   |
+      | http://localhost:8668/v2/entities/urn:ngsi-ld:Device:filling001/attrs/filling?aggrMethod=max&fromDate=2018-06-27T09:00:00&toDate=2050-06-30T23:59:59 |
+      | http://localhost:8668/v2/types/Device/attrs/heartRate?lastN=4&georel=near;maxDistance:5000&geometry=point&coords=52.518,13.357                       |
+      | http://localhost:8668/v2/types/Device/attrs/heartRate?lastN=4&georel=coveredBy&geometry=polygon&coords=52.5537,13.3996;52.5557,13.3996;52.5557,13.3976;52.5537,13.3976;52.5537,13.3996 |
+
+    Scenario Outline: Using Cratedb API
+      When I prepare a POST HTTP request to "http://localhost:4200/_sql"
+      And  I set header content-type to application/json
+      And  I set simple sensor values as described in "<value>"
+      And  I perform the request
+      Then I receive a HTTP "200" response code
+    Examples:
+      | value |
+      | {"stmt":"SHOW SCHEMAS"} |
+      | {"stmt":"SHOW TABLES"}  |
+      | {"stmt":"SELECT * FROM mtopeniot.etFillingLevelSensor WHERE entity_id = 'urn:ngsi-ld:Device:filling001' ORDER BY time_index ASC LIMIT 3"} |
+      | {"stmt":"SELECT * FROM mtopeniot.etFillingLevelSensor WHERE entity_id = 'urn:ngsi-ld:Device:filling001'  ORDER BY time_index DESC LIMIT 3"} |
+      | {"stmt":"SELECT DATE_FORMAT (DATE_TRUNC ('minute', time_index)) AS minute, SUM (filling) AS sum FROM mtopeniot.etFillingLevelSensor WHERE entity_id = 'urn:ngsi-ld:Device:filling001' GROUP BY minute LIMIT 3"} |
+      | {"stmt":"SELECT DATE_FORMAT (DATE_TRUNC ('minute', time_index)) AS minute, MIN (filling) AS min FROM mtopeniot.etFillingLevelSensor WHERE entity_id = 'urn:ngsi-ld:Device:filling001' GROUP BY minute"} |
+      | {"stmt":"SELECT MAX(filling) AS max FROM mtopeniot.etFillingLevelSensor WHERE entity_id = 'urn:ngsi-ld:Device:filling001' and time_index >= '2022-04-01T09:00:00' and time_index < '2025-06-30T23:59:59'"} |
```

Comment:
MAGIC NUMBER or to be precise MAGIC DATE...

I assume that a future execution of the test will fail if we kept these dates...

Reply:
Tutorial uses a "magic number" (date  in this case) -- I've set the date in 2050 to prevent it from failing soon, and I've added a new date with the query until NOW() -- As done in PostgreSQL

Reply:
There's also the same issue in queries to Quantum Leap -- It might fail in 2050. -- However, quantum leap doesn't seem to allow me to use any "now()" function.

Reply:
Now you put one with 2050 and other with now(), just to confirm that this is the expected value and not a mistake.

Reply:
NTC